### ISSUE TITLE
time conversion on travis

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-R/streamMetabolizer#141]
New and interesting failure on Travis today (https://travis-ci.org/USGS-R/streamMetabolizer/builds/104157263): 
```
> test_check('streamMetabolizer')
Loading required package: streamMetabolizer
This information is preliminary or provisional and is subject to revision. It is being provided to meet the need for timely best science. The information has not received final approval by the U.S. Geological Survey (USGS) and is provided on the condition that neither the USGS nor the U.S. Government shall be held liable for any damages resulting from the authorized or unauthorized use of the information. Although this software program has been used by the USGS, no warranty, expressed or implied, is made by the USGS or the U.S. Government as to the accuracy and functioning of the program and related program material nor shall the fact of distribution constitute any such warranty, and no responsibility is assumed by the USGS in connection therewith.
1. Failure (at test-convert.R#35): converting between date and DOY works -------
convert_date_to_doyhr(as.POSIXct("2016-05-29 01:00:00", tz = "CST6CDT")) not equal to 150
150 - 150 == 0.0417
treat numbers as true time since jan 1, ignoring daylight time

2. Failure (at test-convert.R#42): converting between date and DOY works -------
convert_date_to_doyhr(convert_doyhr_to_date(120, year = 1998, tz = "CST6CDT")) not equal to 120
120 - 120 == 0.0417
should preserve DOY in there&back even during daylight savings

3. Failure (at test-convert.R#43): converting between date and DOY works -------
convert_doyhr_to_date(convert_date_to_doyhr(as.POSIXct("2007-06-15 12:00:00", tz = "MST7MDT")), 2007, tz = "MST7MDT") not equal to as.POSIXct("2007-06-15 12:00:00", tz = "MST7MDT")
Mean absolute difference: 3600
preserve date in there&back

Calling 2 simulations using the parallel method...
All chains have finished
Simulation complete.  Reading coda files...
Coda files loaded successfully
Finished running the simulation
Calling 2 simulations using the parallel method...
All chains have finished
Simulation complete.  Reading coda files...
Coda files loaded successfully
Finished running the simulation
testthat results ================================================================
OK: 289 SKIPPED: 0 FAILED: 3
1. Failure (at test-convert.R#35): converting between date and DOY works 
2. Failure (at test-convert.R#42): converting between date and DOY works 
3. Failure (at test-convert.R#43): converting between date and DOY works 

Error: testthat unit tests failed
In addition: Warning messages:
1: In getSchmidt(temperature, gas) : temperature out of range
2: In getSchmidt(temperature, gas) : temperature out of range
3: In getSchmidt(temperature, gas) : temperature out of range
4: In getSchmidt(temperature, gas) : temperature out of range
5: In getSchmidt(temperature, gas) : temperature out of range
6: In getSchmidt(temperature, gas) : temperature out of range
7: In getSchmidt(temperature, gas) : temperature out of range
8: In getSchmidt(temperature, gas) : temperature out of range
9: In getSchmidt(temperature, gas) : temperature out of range
Execution halted

travis_time:end:01faefe6:start=1453490012280903437,finish=1453490012287478155,duration=6574718
[0KR CMD check failed
```

**Comment by aappling-usgs** [comment]
On my computer, ctrl-shift-T (`devtools::test()`) does just fine. Same number of getSchmidt warnings (9), but no failures.

**Comment by aappling-usgs** [comment]
Here's the whole testthat block:
```r
test_that("converting between date and DOY works", {
  # date to doyhr
  expect_equal(convert_date_to_doyhr(as.POSIXct("2020-01-01 00:00:00", tz="GMT")), 1, tol=0.000001, info="Jan 1 should be 1")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2020-01-01 00:00:00", tz="CST6CDT")), 1, tol=0.000001, info="use the same timezone as the arg date")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2020-01-01 00:00:00")), 1, tol=0.000001, info="should work in any tester's default tz")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2020-01-01 01:00:00")), 1+1/24, tol=0.000001, info="decimal should include hours")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2020-01-01 01:03:58")), 1+(60+3+58/60)/(24*60), tol=0.000001, info="decimal should include minutes and seconds")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2004-12-01 00:00:00")), 1+convert_date_to_doyhr(as.POSIXct("2019-12-01 00:00:00")), tol=0.000001, info="should catch leap days")
  expect_equal(convert_date_to_doyhr(as.POSIXct("2016-05-29 01:00:00", tz="CST6CDT")), 150, info="treat numbers as true time since jan 1, ignoring daylight time")
  # doyhr to date
  expect_equal(convert_doyhr_to_date(1, year=1920), as.POSIXct("1920-01-01 00:00:00", tz="GMT"), info="1 should be Jan 1, default GMT")
  expect_equal(convert_doyhr_to_date(3, year=2016, tz="CST6CDT"), as.POSIXct("2016-01-03 00:00:00", tz="CST6CDT"), info="tz should stay as indicated")
  expect_equal(convert_doyhr_to_date(150, year=2016, tz="CST6CDT"), as.POSIXct("2016-05-29 01:00:00", tz="CST6CDT"), info="treat numbers as true time since jan 1, ignoring daylight time")
  # there and back
  expect_equal(convert_date_to_doyhr(convert_doyhr_to_date(12, year=2024, tz="CST6CDT")), 12, info="should preserve DOY in there&back")
  expect_equal(convert_date_to_doyhr(convert_doyhr_to_date(120, year=1998, tz="CST6CDT")), 120, info="should preserve DOY in there&back even during daylight savings")
  expect_equal(convert_doyhr_to_date(convert_date_to_doyhr(as.POSIXct("2007-06-15 12:00:00", tz="MST7MDT")), 2007, tz="MST7MDT"), 
               as.POSIXct("2007-06-15 12:00:00", tz="MST7MDT"), info="preserve date in there&back") 
})
```
and here are the lines that fail:
```r
expect_equal(convert_date_to_doyhr(as.POSIXct("2016-05-29 01:00:00", tz="CST6CDT")), 150, info="treat numbers as true time since jan 1, ignoring daylight time")
expect_equal(convert_date_to_doyhr(convert_doyhr_to_date(120, year=1998, tz="CST6CDT")), 120, info="should preserve DOY in there&back even during daylight savings")
expect_equal(convert_doyhr_to_date(convert_date_to_doyhr(as.POSIXct("2007-06-15 12:00:00", tz="MST7MDT")), 2007, tz="MST7MDT"), 
             as.POSIXct("2007-06-15 12:00:00", tz="MST7MDT"), info="preserve date in there&back") 
```

looks to me like the failure is happening only when daylight savings time is on (summer). this is consistent with the numerical differences in the error messages - 0.0417 is 1/24 day is 1 hour, and 3600 seconds is also 1 hour. so there's something about the (new?) Travis environment that's failing to understand time zones as we're communicating them.

**Comment by aappling-usgs** [comment]
The recent Travis change is that travis-ci.org builds (i.e., ours) are now running on Google Compute Engine (GCE). 

> This job ran on our new platform for Precise builds. Please read [our blog post](https://blog.travis-ci.com/2015-11-27-moving-to-a-more-elastic-future) for more information.

There appear to be two options for `sudo: required` builds: there's Legacy Precise (what I think we've been running based on the above) and Trusty Beta (https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments). I think there's time zone information missing from Legacy Precise. It's possible we can fix that there, but a quick Google isn't turning up the answer. Maybe just running Trusty instead will work?

**Comment by aappling-usgs** [comment]
Nope. Rats.
https://travis-ci.org/USGS-R/streamMetabolizer/builds/106360415

**Comment by aappling-usgs** [comment]
maybe the problem is that CST6CDT and, MST7MDT, etc. are unrecognized time zones in Ubuntu - see http://manpages.ubuntu.com/manpages/saucy/man3/DateTime::TimeZone::Catalog.3pm.html. (Why did these work OK before?? anyway...). I'll try it.

**Comment by aappling-usgs** [comment]
solved in #146 by switching from `lubridate::floor_date` to a `format`-based solution:
```r
 -  out <- 1 + as.numeric(v(date) - floor_date(v(date), unit="year"), units="days") # days_since_dec31
 +  year <- as.POSIXct(format(v(date), "%Y-01-01 00:00:00"), tz=lubridate::tz(v(date)))
 +  out <- as.numeric(v(date) - year, units="days") + 1 # days_since_dec31
```
This seems to work better for Travis (Linux) and still works for my computer (Windows). Don't know about Appveyor (Windows) because of that `install_github` thing =(.

A bug report to `lubridate` could be in order - `floor_date` appears to be mishandling daylight savings on at least one (but not all) versions of Linux. Or this might be Linux's fault rather than `lubridate`'s. Wish I could explore this more directly...
